src/main/java/com/bloomtech/asylumrgbea/services/AsylumCaseService.java

```diff
@@ -1,13 +1,129 @@
 package com.bloomtech.asylumrgbea.services;
 
-import com.bloomtech.asylumrgbea.models.AsylumCaseRequestDto;
-import com.bloomtech.asylumrgbea.models.AsylumCaseResponseDto;
+import com.bloomtech.asylumrgbea.controllers.exceptions.AsylumCaseNotFoundException;
+import com.bloomtech.asylumrgbea.controllers.exceptions.BadRequestException;
+import com.bloomtech.asylumrgbea.entities.AsylumCase;
+import com.bloomtech.asylumrgbea.mappers.AsylumCaseMapper;
+import com.bloomtech.asylumrgbea.models.CaseResponseDto;
+import com.bloomtech.asylumrgbea.models.CasesRequestDto;
+import com.bloomtech.asylumrgbea.models.Page;
 import com.bloomtech.asylumrgbea.models.PageResponseDto;
+import com.bloomtech.asylumrgbea.repositories.AsylumCaseRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
 
-public interface AsylumCaseService {
+import java.util.*;
 
-	PageResponseDto getPageOfAsylumCases(AsylumCaseRequestDto asylumCaseRequestDto);
+@Service
+@RequiredArgsConstructor
+public class AsylumCaseService {
 
-	Iterable<AsylumCaseResponseDto> getAllAsylumCases();
+	private final AsylumCaseRepository asylumCaseRepository;
 
+	private final AsylumCaseMapper asylumCaseMapper;
+
+	/**
+	 * Gets all entities from repository and returns them out of the program.
+	 * @return This is an Iterable of AsylumCase objects.
+	 */
+	@Cacheable("asylum_case_cache")
+	public Iterable<CasesRequestDto> getAllAsylumCases() {
+
+		Iterable<AsylumCase> caseIterable = asylumCaseRepository.findAll();
+		validateIterableIsNotEmpty(caseIterable);
+
+		return asylumCaseMapper.entitiesToResponseDtos(caseIterable);
+	}
+
+	/**
+	 * Gets a page of entities from repository and returns them out of the program.
+	 * @param casesRequestDto Contains the page number as a field.
+	 * @return This is a limited Iterable of AsylumCase objects.
+	 */
+	public PageResponseDto getCasesBy(CasesRequestDto casesRequestDto) {
+		Map<String, List<String>> filterMap = Map.of(
+				"citizenship", 		getListOfFilters(casesRequestDto.getCitizenship(),	"0"),
+				"caseOutcome", 		getListOfFilters(casesRequestDto.getCaseOutcome(),	","),
+				"currentDate", 		getListOfFilters(casesRequestDto.getCurrentDate(),	null),
+				"asylumOffice", 	getListOfFilters(casesRequestDto.getAsylumOffice(), ","));
+
```

Comment:
Double check the variable names, I thought some of them were updated, but maybe that will be done in another ticket.